### PR TITLE
remove create-react-class dependency from main sources

### PR DESCRIPTION
**Pull request by jochenberger**
Basically a rebase of #1702 onto master
see also #1658, #1727

### CODE REVIEW

**Comment by agirton** [comment]
Hi @jochenberger thank you for the contribution! Per the contributing guidelines please update the PR to only include the src files. Thanks!

**Comment by nishp1** [comment]
Any update on this? Would like to get this merged to reduce our bundle size by not adding create class as dep.

**Comment by agirton** [comment]
Hi @jochenberger it looks like this includes a few changes that are not just the conversion ([268](https://github.com/JedWatson/react-select/pull/1734/files#diff-04191649796841200b33375bdff7c78fR268), [872](https://github.com/JedWatson/react-select/pull/1734/files#diff-04191649796841200b33375bdff7c78fL872). [881](https://github.com/JedWatson/react-select/pull/1734/files#diff-04191649796841200b33375bdff7c78fL881), [907](https://github.com/JedWatson/react-select/pull/1734/files#diff-04191649796841200b33375bdff7c78fL907)). Can you remove those and then update.

**Comment by jochenberger** [comment]
I reverted the unrelated changes, sorry.

File: src/Select.js [hunk 1]
```diff
@@ -264,34 +167,40 @@ const Select = createClass({
...
 		}
-	},
+	}
 
 	focus () {
 		if (!this.input) return;
 		this.input.focus();
-	},
+
+		if (this.props.openAfterFocus) {
```
Comment (agirton) [inline src/Select.js]: Looks like there is one last unrelated change 😄

Comment (jochenberger) [inline src/Select.js]: I guess I should have started from scratch rather than trying to port #1702.

File: src/Select.js [hunk 2]
```diff
@@ -1081,7 +984,6 @@ const Select = createClass({
 		let className = classNames('Select', this.props.className, {
 			'Select--multi': this.props.multi,
 			'Select--single': !this.props.multi,
-			'is-clearable': this.props.clearable,
```
Comment (agirton) [inline src/Select.js]: Oops looks like they also removed this as well. 

This PR is starting to make me question if this conversion was done by hand or not 😱 .  @rgrocha how did you generate the code?

Comment (jochenberger) [inline src/Select.js]: I guess it was done by hand but against an older tree. I guess I lost the changes that were made against master since then.

**Comment by jochenberger** [comment]
Is there anything I can do to speed things up? Should I create separate PRs for each file?

**Comment by jochenberger** [comment]
Closing in favor of #1820, #1821, #1822, #1823, and #1824

**Comment by agirton** [comment]
Hi @jochenberger creating multiple PRs makes it harder to deal with. It would be better for one PR with multiple commits.
